cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CaReconcilerTest.java

```diff
@@ -1529,22 +1530,33 @@ public void testRollingReasonsWithClusterCAKeyNotTrusted(Vertx vertx, VertxTestC
         StrimziPodSetOperator spsOps = supplier.strimziPodSetOperator;
         when(spsOps.getAsync(eq(NAMESPACE), eq(KafkaResources.zookeeperComponentName(NAME)))).thenReturn(Future.succeededFuture());
 
+        List<Pod> pods = new ArrayList<>();
+        // adding a terminating Cruise Control pod to test that it's skipped during the key generation check
+        Pod ccPod = podWithNameAndAnnotations("my-cluster-cruise-control", false, false, generationAnnotations);
+        ccPod.getMetadata().setDeletionTimestamp("2023-06-08T16:23:18Z");
+        pods.add(ccPod);
+        // adding Kafka pods with old CA cert and key generation
+        pods.add(podWithNameAndAnnotations("my-cluster-controllers-3", false, true, generationAnnotations));
+        pods.add(podWithNameAndAnnotations("my-cluster-controllers-4", false, true, generationAnnotations));
+        pods.add(podWithNameAndAnnotations("my-cluster-controllers-5", false, true, generationAnnotations));
+        pods.add(podWithNameAndAnnotations("my-cluster-brokers-0", true, false, generationAnnotations));
+        pods.add(podWithNameAndAnnotations("my-cluster-brokers-1", true, false, generationAnnotations));
+        pods.add(podWithNameAndAnnotations("my-cluster-brokers-2", true, false, generationAnnotations));
+
+        StrimziPodSet strimziPodSet = new StrimziPodSetBuilder()
+                .withNewMetadata()
+                    .withName(NAME)
+                .endMetadata()
+                .withNewSpec()
+                    .withPods(PodSetUtils.podsToMaps(pods))
+                .endSpec()
+                .build();
```

Comment:
This seems to be quite wrong:
* The CC Pod would never be in the PodSet
* The Kafka pods would be in two separate pod sets

Reply:
Good point, I'll update it